### PR TITLE
Add consul.xds.server.streamStart metric

### PR DESCRIPTION
**Pull request by pglass**
### Description

This adds a new `consul.xds.server.streamStart` metric to measure the time taken to first generate xDS resources after an xDS stream is opened.

### Testing & Reproduction steps
Unit test updated

<details>
<summary>Manual Test</summary>
No measured values after Consul starts:

```
$ curl -s localhost:8500/v1/agent/metrics?format=prometheus | grep streamStart
# HELP consul_xds_server_streamStart Measures the time in milliseconds after an xDS stream is opened until xDS resources are first generated for the stream.
# TYPE consul_xds_server_streamStart summary
consul_xds_server_streamStart{quantile="0.5"} NaN
consul_xds_server_streamStart{quantile="0.9"} NaN
consul_xds_server_streamStart{quantile="0.99"} NaN
consul_xds_server_streamStart_sum 0
consul_xds_server_streamStart_count 0
```

After one consul-dataplane has started:

```
$ curl -s localhost:8500/v1/agent/metrics?format=prometheus | grep streamStart
# HELP consul_xds_server_streamStart Measures the time in milliseconds after an xDS stream is opened until xDS resources are first generated for the stream.
# TYPE consul_xds_server_streamStart summary
consul_xds_server_streamStart{quantile="0.5"} 207
consul_xds_server_streamStart{quantile="0.9"} 207
consul_xds_server_streamStart{quantile="0.99"} 207
consul_xds_server_streamStart_sum 207
consul_xds_server_streamStart_count 1
```

After consul-dataplane has started a few times:

```
$ curl -s localhost:8500/v1/agent/metrics?format=prometheus | grep streamStart
# HELP consul_xds_server_streamStart Measures the time in milliseconds after an xDS stream is opened until xDS resources are first generated for the stream.
# TYPE consul_xds_server_streamStart summary
consul_xds_server_streamStart{quantile="0.5"} 201
consul_xds_server_streamStart{quantile="0.9"} 201
consul_xds_server_streamStart{quantile="0.99"} 201
consul_xds_server_streamStart_sum 609
consul_xds_server_streamStart_count 3
```

</details>

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [x] not a security concern

### CODE REVIEW

File: agent/xds/delta.go [hunk 1]
```diff
@@ -360,6 +364,11 @@ func (s *Server) processDelta(stream ADSDeltaStream, reqCh <-chan *envoy_discove
 
 			generator.Logger.Trace("Invoking all xDS resource handlers and sending changed data if there are any")
 
+			streamStartOnce.Do(func() {
+				diff := time.Since(streamStartTime)
+				metrics.AddSample([]string{"xds", "server", "streamStart"}, float32(diff/time.Millisecond))
```
Comment (boxofrad) [inline agent/xds/delta.go:369]: Is there a specific reason not to use [`MeasureSince`](https://pkg.go.dev/github.com/armon/go-metrics#MeasureSince) here?

**Review by boxofrad — COMMENTED** [review]
Looks good to me! Just one question about the choice of metrics method.

File: agent/xds/delta.go (hunk 1 above)
Comment (pglass) [reply agent/xds/delta.go]: Oh, that's perfect. Fixed!

**Review by boxofrad — APPROVED** [review]

**Review by trujillo-adam — APPROVED** [review]
Looks good from a the docs perspective.

**Pull request merged**